File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        Arrays.sort(nums);
        int i=0, expected=nums[i], seqCounter=0, result=0, len=nums.length;
        while (i < len) {
            if (expected != nums[i]) {
                expected = nums[i];
                seqCounter = 0;
            }
            while (i < len && expected == nums[i]) {
                i++;
            }
            seqCounter++;
            expected++;
            result = Math.max(result, seqCounter);
        }
        return result;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        // Step 1 : create map
        const result = nums.reduce((acc ,item) => {
            if(acc[item]){
                acc[item]++
            }
            else{
                acc[item] = 1
            }
            return acc
        } ,{})

        // Step 2 : sort
        const sorted = Object.entries(result).sort((a , b) => b[1] - a[1])
        
        console.log(sorted)

        // Step 3 - get the first k
        return sorted.slice(0, k).map(entry => Number(entry[0])); // Convert keys to numbers
    
    }
}
